src/main/java/cuchaz/enigma/translation/mapping/serde/ProguardMappingsReader.java

```diff
@@ -0,0 +1,135 @@
+package cuchaz.enigma.translation.mapping.serde;
+
+import com.google.common.base.Charsets;
+import cuchaz.enigma.ProgressListener;
+import cuchaz.enigma.command.MappingCommandsUtil;
+import cuchaz.enigma.throwables.MappingParseException;
+import cuchaz.enigma.translation.mapping.EntryMapping;
+import cuchaz.enigma.translation.mapping.MappingSaveParameters;
+import cuchaz.enigma.translation.mapping.tree.EntryTree;
+import cuchaz.enigma.translation.mapping.tree.HashEntryTree;
+import cuchaz.enigma.translation.representation.MethodDescriptor;
+import cuchaz.enigma.translation.representation.TypeDescriptor;
+import cuchaz.enigma.translation.representation.entry.ClassEntry;
+import cuchaz.enigma.translation.representation.entry.FieldEntry;
+import cuchaz.enigma.translation.representation.entry.MethodEntry;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ProguardMappingsReader implements MappingsReader {
+    public static final ProguardMappingsReader INSTANCE = new ProguardMappingsReader();
+    private static final String NAME = "[a-zA-Z0-9_\\-.$<>]+";
+    private static final String TYPE = NAME + "(?:\\[])*";
+    private static final String TYPE_LIST = "|(?:(?:" + TYPE + ",)*" + TYPE + ")";
+    private static final Pattern CLASS = Pattern.compile("(" + NAME + ") -> (" + NAME + "):");
+    private static final Pattern FIELD = Pattern.compile(" {4}(" + TYPE + ") (" + NAME + ") -> (" + NAME + ")");
+    private static final Pattern METHOD = Pattern.compile(" {4}(?:[0-9]+:[0-9]+:)?(" + TYPE + ") (" + NAME + ")\\((" + TYPE_LIST + ")\\) -> (" + NAME + ")");
+
+    public ProguardMappingsReader() {}
+
+
+    @Override
+    public EntryTree<EntryMapping> read(Path path, ProgressListener progress, MappingSaveParameters saveParameters) throws MappingParseException, IOException {
+        EntryTree<EntryMapping> mappings = new HashEntryTree<>();
+
+        int lineNumber = 0;
+        ClassEntry currentClass = null;
+        for (String line : Files.readAllLines(path, Charsets.UTF_8)) {
```

Comment:
StandardCharsets.UTF_8

Reply:
This line is copied from the Enigma mapping reader. I think it makes sense to use Apache's `Charsets` if we're also using its `Files` rather than mixing APIs.

Reply:
Hmm, we are using jdk Files and guava Charsets